### DISCUSSION TITLE
Disable paste rules for specific marks/nodes in v2

### DISCUSSION THREAD

**Discussion by emanuelgsouza** [ueberdosis/tiptap#1524] [General Questions]
I saw that in the v1 we have a merged PR with this feature ( #272 ), but this is not implemented in v2. For now, we have the `enablePasteRules` boolean, but, in my case, I need to disable for specific marks/nodes. This is possible?

**Comment by hanspagel** [comment]
You can overwrite the paste rules with an empty array.

Bold.extend({ addPasteRules() { return [] } )

**Reply by emanuelgsouza** [reply]
It seems this is not working. I will add more context.

I'm creating a richtext editor for our application and we have the functionality to allow to user a way to mark a specific text with a specific class. The mark only renders a span tag around the text.

My question is related the fact when a user copy and past a content from Word, for example, every text is marked with this mark. I would like to know how can I make the editor ignores this mark on paste content.

**Reply by emanuelgsouza** [reply]
I am using the `@tiptap/vue-2` version 2.0.0-beta.32 and `@tiptap/core` version 2.0.0-beta.81.

**Reply by hanspagel** [reply]
You should have a look at parseHTML:

https://www.tiptap.dev/guide/custom-extensions#parse-html

✌️

**Reply by emanuelgsouza** [reply]
Ok, but, what do you mean?

**Reply by hanspagel** [reply]
`parseHTML` parses the pasted content from Word, e. g. when you copy `<span style="font-weight: bold">foo</span>` it’ll be translated to a `Bold` node.

If that doesn’t help, would you mind to reframe your question?

**Reply by emanuelgsouza** [reply]
I will share the code of the mark that we want to ignore on pasting:

```js
import { Mark } from '@tiptap/core'

export default Mark.create({
  name: 'styled',

  addAttributes() {
    return {
      class: {
        default: '',
      },
    }
  },

  parseHTML() {
    return [
      {
        tag: 'span',
      },
    ]
  },

  renderHTML({ HTMLAttributes }) {
    const attrs = {
      class: 'ProseMirror__styled',
      title: HTMLAttributes.class,
    }

    return ['span', attrs, 0]
  },

  addCommands() {
    return {
      resetStyled: () => ({ commands }) => {
        return commands.unsetMark('styled')
      },

      setStyled: (value) => ({ commands }) => {
        return commands.setMark('styled', value)
      },
    }
  },
})
```

**Reply by emanuelgsouza** [reply]
Maybe the following code:

```js
parseHTML() {
    return [
      {
        tag: 'span',
      },
    ]
  },
```

Parses each span to a styled mark in our case, maybe we should only parse the span with the correct class and title attributes, alright?

**Reply by emanuelgsouza** [reply]
@hanspagel I changed the parseHTML function to this:

```js
parseHTML() {
    return [
      {
        tag: 'span',
        getAttrs: (node) =>
          node.getAttribute('class') === 'ProseMirror__styled',
      },
    ]
  },
```

And it worked! Thanks for the help, I could ignore the pasting content here for this mark.
